### PR TITLE
CUI-7368 AlertDialog should have support for labelledBy and describedBy properties

### PR DESCRIPTION
**Pull request by majornista**
## Description
As per discussion in CQ-4292426 [0], coral-dialog component [1] needs support for aria-labelledBy and aria-describedBy properties, at least for the alertDialog so that the title of the dialog and description present in the body can be announced by screen readers.

[0] https://jira.corp.adobe.com/browse/CQ-4292426?focusedCommentId=21075521&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-21075521
[1] https://git.corp.adobe.com/Coral/coralui-component-dialog/blob/master/scripts/Coral.Dialog.js

## Related Issue
https://jira.corp.adobe.com/browse/CUI-7368, https://jira.corp.adobe.com/browse/CQ-4292426

## Motivation and Context
Improve screen reader announcement of `role=alertdialog` variants when the Cancel or Ok button receives focus.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by majornista** [comment]
@adobe/coral-spectrum @icaraps, @mohiaror, @indra2gurjar Could I get a review?

File: coral-component-autocomplete/src/tests/test.Autocomplete.js [hunk 1]
```diff
@@ -1945,7 +1945,9 @@ describe('Autocomplete', function() {
 
           el.on('coral-overlay:close', function() {
             helpers.next(function() {
-              expect(trigger.getAttribute('aria-expanded')).to.equal('false');
+              if (trigger) {
```
Comment (icaraps) [inline coral-component-autocomplete/src/tests/test.Autocomplete.js:1948]: Is this one a flaky test ?

Comment (majornista) [reply coral-component-autocomplete/src/tests/test.Autocomplete.js:1948]: It's a flaky test. My fix is just to test for `trigger` before trying to evaluate.

Comment (icaraps) [reply coral-component-autocomplete/src/tests/test.Autocomplete.js:1948]: You can remove it in favor of https://github.com/adobe/coral-spectrum/commit/1c38579cdd9b1d941be35313c430c0c20f3f5aad

**Pull request merged**